### PR TITLE
PUB-1778 Refactor account controller and account service

### PR DESCRIPTION
**Pull request by KianKwa**
### JIRA link (if applicable) ###

https://tools.hmcts.net/jira/browse/PUB-1778

### Change description ###

Refactor account controller and account service. The AccountController class is split into the following classes:
- AccountController
- AzureAccountController
- BulkAccountCreationController
- CustomAccountRetrieverController
- InactiveAccountManagementController
- SystemAdminAccountController

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/reform/pip/account/management/service/CustomAccountRetrievalService.java [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+import uk.gov.hmcts.reform.pip.account.management.model.UserProvenances;
+
+import java.util.List;
+
+import static uk.gov.hmcts.reform.pip.account.management.model.Roles.ALL_NON_RESTRICTED_ADMIN_ROLES;
+import static uk.gov.hmcts.reform.pip.account.management.model.Roles.ALL_NON_THIRD_PARTY_ROLES;
+import static uk.gov.hmcts.reform.pip.account.management.model.UserProvenances.ALL_NON_THIRD_PARTY_PROVENANCES;
+
+@Service
+public class CustomAccountRetrievalService {
```
Comment (ChrisS1512) [inline src/main/java/uk/gov/hmcts/reform/pip/account/management/service/CustomAccountRetrievalService.java:21]: Similar comment to the controller name

**Review by ChrisS1512 — CHANGES_REQUESTED** [review]

File: src/main/java/uk/gov/hmcts/reform/pip/account/management/validation/validator/NotSystemAdminValidatorAzureAccount.java [hunk 2]
```diff
@@ -1,20 +1,20 @@
...
 /**
  * Validator class that checks if an azure account is a system admin.
  */
-@SuppressWarnings({"PMD.LawOfDemeter"})
 public class NotSystemAdminValidatorAzureAccount implements ConstraintValidator<NotSystemAdmin, AzureAccount> {
 
     @Override
     public boolean isValid(AzureAccount azureAccount, ConstraintValidatorContext constraintValidatorContext) {
-        return !Roles.SYSTEM_ADMIN.equals(azureAccount.getRole());
+        return !SYSTEM_ADMIN.equals(azureAccount.getRole());
```
[the remark this answers is not in the document]
Comment (KianKwa) [reply src/main/java/uk/gov/hmcts/reform/pip/account/management/validation/validator/NotSystemAdminValidatorAzureAccount.java:18]: Yeah should clean up those suppressions everywhere in the project! Annoying PMD think this is probably a false positive complaints

File: src/main/java/uk/gov/hmcts/reform/pip/account/management/service/CustomAccountRetrievalService.java (hunk 1 above)
Comment (KianKwa) [reply src/main/java/uk/gov/hmcts/reform/pip/account/management/service/CustomAccountRetrievalService.java]: This one is renamed to AccountFilteringService :)

File: src/main/java/uk/gov/hmcts/reform/pip/account/management/controllers/CustomAccountRetrievalController.java [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+import uk.gov.hmcts.reform.pip.account.management.service.CustomAccountRetrievalService;
+
+import java.util.List;
+
+@RestController
+@Tag(name = "Account Management - API for retrieving custom user accounts")
+@RequestMapping("/account")
+@Validated
+@IsAdmin
+public class CustomAccountRetrievalController {
```
[the remark this answers is not in the document]
Comment (KianKwa) [reply src/main/java/uk/gov/hmcts/reform/pip/account/management/controllers/CustomAccountRetrievalController.java]: Spoke with Danny and rename this to AccountFilteringController

**Review by ChrisS1512 — CHANGES_REQUESTED** [review]
Looks good :) Just got a conflict but once sorted happy to approve

**Review by ChrisS1512 — APPROVED** [review]

**Review by joshblackmoor — APPROVED** [review]

**Pull request merged**